src/Database/TigerBeetle/Internal/FFI/Account.hsc

```diff
@@ -61,21 +62,21 @@ instance Storable TBAccount where
 
     alignment _ = #{alignment tb_account_t}
 
-    peek ptr
-      = TBAccount
-      <$> #{peek tb_account_t, id} ptr
-      <*> #{peek tb_account_t, debits_pending} ptr
-      <*> #{peek tb_account_t, debits_posted} ptr
-      <*> #{peek tb_account_t, credits_pending} ptr
-      <*> #{peek tb_account_t, credits_posted} ptr
-      <*> #{peek tb_account_t, user_data_128} ptr
-      <*> #{peek tb_account_t, user_data_64} ptr
-      <*> #{peek tb_account_t, user_data_32} ptr
-      <*> #{peek tb_account_t, reserved} ptr
-      <*> #{peek tb_account_t, ledger} ptr
-      <*> #{peek tb_account_t, code} ptr
-      <*> #{peek tb_account_t, flags} ptr
-      <*> #{peek tb_account_t, timestamp} ptr
+    peek ptr = do
+      tbAccountId             <- #{peek tb_account_t, id} ptr
+      tbAccountDebitsPending  <- #{peek tb_account_t, debits_pending} ptr
+      tbAccountDebitsPosted   <- #{peek tb_account_t, debits_posted} ptr
+      tbAccountCreditsPending <- #{peek tb_account_t, credits_pending} ptr
+      tbAccountCreditsPosted  <- #{peek tb_account_t, credits_posted} ptr
+      tbAccountUserData128    <- #{peek tb_account_t, user_data_128} ptr
+      tbAccountUserData64     <- #{peek tb_account_t, user_data_64} ptr
+      tbAccountUserData32     <- #{peek tb_account_t, user_data_32} ptr
+      tbAccountReserved       <- #{peek tb_account_t, reserved} ptr
+      tbAccountLedger         <- #{peek tb_account_t, ledger} ptr
+      tbAccountCode           <- #{peek tb_account_t, code} ptr
+      tbAccountFlags          <- #{peek tb_account_t, flags} ptr
+      tbAccountTimestamp      <- #{peek tb_account_t, timestamp} ptr
+      pure TBAccount{..}
```

Comment:
We should avoid rewriting code that doesn't change behavior as it adds more code to review that doesn't do anything.

Reply:
Oh yeah, I agree about code that doesn't change behaviour, but IMO this changes the way that code can evolve. By not depending on the order of fields you are free to re-order fields in the record, or add fields in different places, and you will get good errors if fields change.